code/game/objects/effects/decals/turfdecal/tilecoloring.dm

```diff
@@ -6,19 +6,39 @@
 
 #define PRIDE_ALPHA 60
 
+GLOBAL_LIST_INIT(rainbow_colors, list(
+		COLOR_PRIDE_PURPLE,
+		COLOR_PRIDE_BLUE,
+		COLOR_PRIDE_GREEN,
+		COLOR_PRIDE_YELLOW,
+		COLOR_PRIDE_ORANGE,
+		COLOR_PRIDE_RED,
+	))
+
 /obj/effect/turf_decal/tile/Initialize(mapload)
-	if (check_holidays(APRIL_FOOLS))
+	if(check_holidays(APRIL_FOOLS))
 		color = "#[random_short_color()]"
-	else if (check_holidays(PRIDE_WEEK))
-		var/datum/holiday/pride_week/pride_week = GLOB.holidays[PRIDE_WEEK]
-		color = pride_week.get_floor_tile_color(src)
 
+	else if(check_holidays(PRIDE_WEEK))
+		color = pride_recolor(src)
 		// It looks garish at different alphas, and it's not possible to get a
 		// consistent color palette without this.
 		alpha = PRIDE_ALPHA
 	return ..()
 
-#undef PRIDE_ALPHA
+/// Given an atom, will return what color it should be to match the pride flag.
+/obj/effect/turf_decal/proc/pride_recolor(atom/atom)
+	var/turf/turf = get_turf(atom)
+	return GLOB.rainbow_colors[(turf.y % GLOB.rainbow_colors.len) + 1]
+
+/// Tram needs x-axis aligned tiles
+/obj/effect/turf_decal/tile/neutral/tram/pride_recolor(atom/atom)
+	var/turf/turf = get_turf(atom)
+	return GLOB.rainbow_colors[(turf.x % GLOB.rainbow_colors.len) + 1]
+
+/obj/effect/turf_decal/trimline/tram/pride_recolor(atom/atom)
+	var/turf/turf = get_turf(atom)
```

Comment:
You shouldn't need this, every atom has x/y/z vars

Reply:
Yep, removed.